spinoso-time/src/time/tzrs/mod.rs

```diff
@@ -0,0 +1,259 @@
+use core::cmp::Ordering;
+use core::hash::{Hash, Hasher};
+
+use tz::datetime::DateTime;
+
+mod build;
+mod convert;
+mod math;
+mod offset;
+mod parts;
+mod timezone;
+mod to_a;
+
+pub use offset::Offset;
+pub use to_a::ToA;
+
+use crate::NANOS_IN_SECOND;
+
+/// A wrapper around [`tz::datetime::DateTime`] which contains everything needed for date creation and
+/// conversion to match the ruby spec. Seconds and Subseconds are stored independently as i64 and
+/// u32 respectively, which gives enough granularity to meet the ruby [`Time`] spec.
+///
+/// [`Time`]: https://ruby-doc.org/core-2.6.3/Time.html
+#[must_use]
+#[derive(Debug, Copy, Clone)]
+pub struct Time {
+    /// A wrapper around [`tz::datetime::DateTime`] to provide date and time formatting
+    inner: DateTime,
+    /// The offset to used for the provided _time_
```

Comment:
sorry for these nits

```suggestion
    /// The offset to used for the provided _time_.
```